pig/src/main/java/com/netflix/iceberg/pig/IcebergInputFormat.java

```diff
@@ -0,0 +1,260 @@
+    scan.planTasks().forEach((scanTask) -> splits.add(new IcebergSplit(scanTask)));
+
+    return splits;
+  }
+
+  @Override
+  public RecordReader<Void, T> createRecordReader(InputSplit split, TaskAttemptContext context) throws IOException, InterruptedException {
+    return new IcebergRecordReader<>();
+  }
+
+  public static class IcebergSplit extends InputSplit implements Writable {
+    private CombinedScanTask task;
+
+    public IcebergSplit() {
+
+    }
+
+    public IcebergSplit(CombinedScanTask task) {
+      this.task = task;
+    }
+
+    @Override
+    public long getLength() throws IOException, InterruptedException {
+      return task.files().stream().mapToLong(FileScanTask::length).sum();
+    }
+
+    @Override
+    public String[] getLocations() throws IOException, InterruptedException {
+      return new String[0];
+    }
+
+    @Override
+    public void write(DataOutput out) throws IOException {
+      byte[] data = SerializationUtils.serialize(this.task);
+      out.writeInt(data.length);
+      out.write(data);
+    }
+
+    @Override
+    public void readFields(DataInput in) throws IOException {
+      byte[] data = new byte[in.readInt()];
+      in.readFully(data);
+
+      this.task = (CombinedScanTask) SerializationUtils.deserialize(data);
+    }
+  }
+
+  public class IcebergRecordReader<T> extends RecordReader<Void, T> {
+    private IcebergSplit split;
+    private TaskAttemptContext context;
+    private ReadSupport<T> readSupport;
+
+    private Iterator<FileScanTask> tasks;
+    private FileScanTask currentTask;
+
+    private Iterator<T> recordIterator;
+    private T currentRecord;
+
+    @Override
+    public void initialize(InputSplit split, TaskAttemptContext context) throws IOException, InterruptedException {
+      this.split = (IcebergSplit) split;
+      this.context = context;
+
+      CombinedScanTask ctask = ((IcebergSplit) split).task;
+      tasks = ctask.files().iterator();
+
+      advance();
+    }
+
+    @SuppressWarnings("unchecked")
+    private boolean advance() throws IOException {
+      if (!tasks.hasNext()) {
+        return false;
+      }
+      currentTask = tasks.next();
+
+      Schema tableSchema = (Schema) ObjectSerializer.deserialize(context.getConfiguration().get(ICEBERG_SCHEMA));
+      List<String> projectedFields = (List<String>) ObjectSerializer.deserialize(context.getConfiguration().get(ICEBERG_PROJECTED_FIELDS));
+
+      Schema projectedSchema = projectedFields != null ? project(tableSchema, projectedFields) : tableSchema;
+
+      PartitionSpec spec = currentTask.asFileScanTask().spec();
+      DataFile file = currentTask.file();
+      InputFile inputFile = HadoopInputFile.fromLocation(file.path(), context.getConfiguration());
+
+      Set<Integer> idColumns = spec.identitySourceIds();
+
+      // schema needed for the projection and filtering
+      boolean hasJoinedPartitionColumns = !idColumns.isEmpty();
+
+      switch (file.format()) {
+        case PARQUET:
```

Comment:
If this only supports Parquet, then we should open a follow-up issue to add Avro support.